app/models/mixins/relationship_mixin.rb

```diff
@@ -250,6 +250,48 @@ def child_rels(*args)
     Relationship.filter_by_resource_type(rels, options)
   end
 
+  # depth = depth + 2 ()
+  #
+  # ancestry:
+  #   my children:              ancestry == me.child_ancestry
+  #   my grand children:        ancestry == "#{me.child_ancestry}/[0-9]*"
+  #   my grand grand children:  ancestry == "#{me.child_ancestry}/[0-9]*/[0-9]*"
+  #
+  # regexp was simpler but 10x slower. so used like
+  #
+  # algorithm:
+  #
+  # relationships.each
+  #   rels so far OR                                        -- AR or (not an arel or)
+  #       (relationship LIKE "child_ancestry/%"             -- grand child or lower
+  #       AND NOT relationship LIKE "child_ancestry/%/%")   -- grand grand child or lower
+  #
+  # reminders:
+  # - matches("a%", nil, true) is case sensitive matching (i.e.: "like 'a%'") vs default (i.e.: "ilike 'a%'")
+  #
+  def grand_child_rels(*args)
+    options = args.extract_options!
+    pk = Relationship.arel_table[Relationship.ancestry_column]
+    rels = relationships.inject(Relationship.none) do |stmt, r|
+             stmt.or(Relationship.where(pk.matches("#{r.child_ancestry}/%", nil, true)
+                                          .and(pk.does_not_match("#{r.child_ancestry}/%/%", nil, true))))
```

Comment:
working on grand children right now.

the biggest win in this whole PR is the `nil, true)` here - the rest does drop it by 1/2